jobs/tests/test_lamdba_s3_event.py

```diff
@@ -65,6 +65,13 @@ def setUp(self):
         self.mockresponses.start()
         for url, payload in self.remote_content.items():
             self.mockresponses.get(utils.ARCHIVE_URL + url, payload=payload)
+        # This will be attempted every time since the metadata cache_folder
+        # is always reset by the pytest fixture.
+        self.mockresponses.get(
+            utils.ARCHIVE_URL + 'pub/firefox/nightly/2016/05/2016-05-02-03-02'
+            '-07-mozilla-central/firefox-49.0a1.en-US.linux-i686.json',
+            status=404
+        )
```

Comment:
I find weird to have this here honestly.  For `CSVToRecords` it seems ok, because the remote_content is defined on the same class. But here, it's the parent class, I think we can do better :) 

What about:

```py
    for url, payload in self.remote_content.items():
        if instanceof(payload, int):
            self.mockresponses.get(utils.ARCHIVE_URL + url, status=payload)
        else:
            self.mockresponses.get(utils.ARCHIVE_URL + url, payload=payload)
```

And put this `firefox-49.0a1.en-US.linux-i686` where it belongs?

Reply:
You're right! It's clear now when I "zoom out".